### PR TITLE
Update / Remove Deprecated Code

### PR DESCRIPTION
**Pull request by Noahbg**
Now Uses self:GetOwner() instead of self.Owner.

### CODE REVIEW

File: garrysmod/gamemodes/terrortown/entities/effects/teleport_beamdown.lua [hunk 1]
```diff
@@ -8,7 +8,7 @@ local mid = 32
 local final_height = top
 
 function EFFECT:Init(data)
-   self.Owner = data:GetEntity()
+   self:GetOwner() = data:GetEntity()
```
Comment (GitSparTV) [inline garrysmod/gamemodes/terrortown/entities/effects/teleport_beamdown.lua:11]: Haha, you just Ctrl+H everything

File: garrysmod/gamemodes/terrortown/entities/effects/teleport_beamdown.lua [hunk 2]
```diff
@@ -23,8 +23,8 @@ function EFFECT:Init(data)
 
    self.BeamDown = false
 
-   if IsValid(self.Owner) then
-      self.Dummy = ClientsideModel(self.Owner:GetModel(), RENDERGROUP_OPAQUE)
+   if IsValid(self:GetOwner()) then
+      self.Dummy = ClientsideModel(self:GetOwner():GetModel(), RENDERGROUP_OPAQUE)
```
Comment (GitSparTV) [inline garrysmod/gamemodes/terrortown/entities/effects/teleport_beamdown.lua:27]: Iirc effects doesn't have GetOwner, they still use .Owner
Needs confirmation

File: garrysmod/lua/includes/extensions/entity.lua [hunk 3]
```diff
@@ -27,7 +27,7 @@ function meta:__index( key )
 	end
 
 	--
-	-- Legacy: sometimes use self.Owner to get the owner.. so lets carry on supporting that stupidness
+	-- Legacy: sometimes use self:GetOwner() to get the owner.. so lets carry on supporting that stupidness
```
Comment (GitSparTV) [inline garrysmod/lua/includes/extensions/entity.lua:30]: Man, sometimes you need to manually look at what you replaced

**Comment by robotboy655** [comment]
1) Separate TTT changes
2) This is a pointless change. self.Owner is too convenient IMO.

**Comment by Kefta** [comment]
self.Owner calls __index an extra time, this should absolutely be changed.

**Pull request closed without merging**